### PR TITLE
add az pipepline yaml

### PR DESCRIPTION
**Pull request by tg123**
First step to enable Azure Pipeline
Example result
<https://farmer1992.visualstudio.com/opensources/_build/results?buildId=270>

 * #16 Add coverage report (via coverlet)
 * #37 Add Resharper code style checker (not enforce)
 * #187 Azure pipeline
 * #239 targeting net452

Built on Windows, as a result, add targeting `net452` to nupkg.
However, `xamarinios10`, `monoandroid81` still cannot be included due to that ios/android sdk are not shipped with azure pipeline image.

Need to create an instance for this report to enable azure pipeline

### CODE REVIEW

File: src/KubernetesClient/KubernetesClient.csproj [hunk 1]
```diff
@@ -9,7 +9,8 @@
     <PackageIconUrl>https://raw.githubusercontent.com/kubernetes/kubernetes/master/logo/logo.png</PackageIconUrl>
     <PackageTags>kubernetes;docker;containers;</PackageTags>
 
-    <TargetFrameworks>netstandard1.4;net452;netcoreapp2.1;xamarinios10;monoandroid81</TargetFrameworks>
+    <TargetFrameworks>netstandard1.4;net452;netcoreapp2.1</TargetFrameworks>
+    <!-- <TargetFrameworks>netstandard1.4;net452;netcoreapp2.1;xamarinios10;monoandroid81</TargetFrameworks> -->
```
Comment (brendandburns) [inline src/KubernetesClient/KubernetesClient.csproj]: There are still people interested in this for xamarin, can we retain it for xamarin?

File: src/KubernetesClient/KubernetesClient.csproj [hunk 2]
```diff
@@ -20,7 +21,6 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.JsonPatch" Version="1.1.2" />
-    <PackageReference Include="Nerdbank.GitVersioning" Version="2.2.33" PrivateAssets="all" />
```
Comment (brendandburns) [inline src/KubernetesClient/KubernetesClient.csproj]: Why is this removed?

File: src/KubernetesClient/KubernetesClient.csproj (hunk 1 above)
Comment (tg123) [reply src/KubernetesClient/KubernetesClient.csproj]: Azure pipeline windows image does not have ios sdk on it. Build will fail only on CI.
My idea is to create another issue for it and ping Azure pipeline team.

File: src/KubernetesClient/KubernetesClient.csproj (hunk 2 above)
Comment (tg123) [reply src/KubernetesClient/KubernetesClient.csproj]: change to azure pipeline versioning

```
  - task: DotNetCoreCLI@2
    displayName: 'dotnet pack'
    inputs:
      command: pack
      packagesToPack: src/KubernetesClient/KubernetesClient.csproj
      packDirectory: '$(Build.ArtifactStagingDirectory)/nupkg'
      majorVersion: 1
      minorVersion: 4      
      versioningScheme: byPrereleaseNumber
```

```
 this will generate a SemVer -compliant version formatted as X.Y.Z-ci-datetime where you choose X, Y, and Z.
```

The pro for generating version when CI is that it differentiate `build` and `commit` .
Same `commit` might be different due to `build` environment changed.

**Comment by brendandburns** [comment]
I don't want to remove Xamarin support from the core project, perhaps we can use a script or conditional to adjust things in the pipeline build?

**Comment by tg123** [comment]
@brendanburns fixed with conditional build. pipeline on azure will yield `net452` `netstandard14` `netcore2.1`

**Comment by RohanNagar** [comment]
You should also be able to build the Xamarin in Azure Pipelines, with an additional macOS build pipeline - https://docs.microsoft.com/en-us/azure/devops/pipelines/languages/xamarin?view=azure-devops

**Comment by brendandburns** [comment]
That sounds good. Let's add a Mac build to the pipeline.

**Comment by tg123** [comment]
If let me choose, I will choose netcore and net452 in our published nuget
and after some test, vsts macos image cannot build ios and android either.

package matrix

WinNT
- [x] netcoreapp2.1
- [x] netstandard1.4
- [x] net452
- [ ] xamarinios10
- [ ] monoandroid81

MacOS
- [x] netcoreapp2.1
- [x] netstandard1.4
- [ ] net452
- [ ] xamarinios10
- [ ] monoandroid81

Ubuntu
- [x] netcoreapp2.1
- [x] netstandard1.4
- [ ] net452
- [ ] xamarinios10
- [ ] monoandroid81

build using mac os
<https://farmer1992.visualstudio.com/opensources/_build/results?buildId=273>

**Comment by brendandburns** [comment]
/lgtm
/approve

@tg123 what do I need to do to set this up for real?

**Comment by brendandburns** [comment]
@tg123 I finally got this setup started, but coverage is failing:

https://dev.azure.com/kubernetes-client/csharp-build/_build/results?buildId=5

Any ideas?

Thanks!

**Comment by tg123** [comment]
@brendandburns 

<https://dev.azure.com/kubernetes-client/csharp-build/_build/results?buildId=6>

👍 

we are programming by accident :P
